src/cript/api/api.py

```diff
@@ -997,11 +948,57 @@ def delete_node_by_uuid(self, node_type: str, node_uuid: str) -> None:
         -------
         None
         """
-        delete_node_api_url: str = f"{self._host}/{node_type.lower()}/{node_uuid}/"
 
-        response: Dict = requests.delete(headers=self._http_headers, url=delete_node_api_url, timeout=_API_TIMEOUT).json()
+        response: Dict = self._capsule_request(url_path=f"/{node_type.lower()}/{node_uuid}/", method="DELETE").json()
 
         if response["code"] != 200:
-            raise APIError(api_error=str(response), http_method="DELETE", api_url=delete_node_api_url)
+            raise APIError(api_error=str(response), http_method="DELETE", api_url=f"/{node_type.lower()}/{node_uuid}/")
 
         self.logger.info(f"Deleted '{node_type.title()}' with UUID of '{node_uuid}' from CRIPT API.")
+
+    def _capsule_request(self, url_path: str, method: str, api_request: bool = True, headers: Optional[Dict] = None, timeout: int = _API_TIMEOUT, **kwargs) -> requests.Response:
```

Comment:
ok - this is the heart of the PR

Reply:
basically would you say with this change brought in I should use capsule each time instead of any manual requests.post/patch/delete

Reply:
yes, that is what it means.

It just makes debugging so much easier, because we can log the requests and send them to the back end team directly, without manually finding and reproducing the error.